code/datums/action.dm

```diff
@@ -579,6 +581,13 @@
 		return spell.can_cast(owner)
 	return FALSE
 
+/datum/action/spell_action/should_show_cooldown()
+	var/obj/effect/proc_holder/spell/S = target
+	var/datum/spell_cooldown/charges/handler = S.cooldown_handler
+	if(!istype(S) || !istype(handler))
+		return ..()
+	return handler.current_charges < handler.max_charges
```

Comment:
Same as below. This is logic that should be delegated to the handler instead of doing `istype` checks to see what kind of handler it is.